core/grant/service_test.go

```diff
@@ -1015,3 +1015,19 @@ func (s *ServiceTestSuite) TestGetGrantsTotalCount() {
 		s.NoError(actualError)
 	})
 }
+
+func (s *ServiceTestSuite) TestListUserRoles() {
+	s.Run("should return error if got error from repository", func() {
+		s.setup()
+		expectedError := errors.New("repository error")
+		s.mockRepository.EXPECT().
+			ListUserRoles(mock.AnythingOfType("*context.emptyCtx"), "user").
+			Return(nil, expectedError).Once()
+
+		actualCount, actualError := s.service.ListUserRoles(context.Background(), "user")
+
+		s.Zero(actualCount)
+		s.EqualError(actualError, expectedError.Error())
+	})
+
```

Comment:
need to add other test cases:
1. successful scenario
2. test case if owner is empty